src/charm.py

```diff
@@ -69,20 +76,24 @@ def _configure_sdcore_nms(self, event: EventBase) -> None:
             self.unit.status = WaitingStatus("Waiting for container to be ready")
             event.defer()
             return
-        for required_relation in [FIVEG_N4_RELATION_NAME, SDCORE_MANAGEMENT_RELATION_NAME]:
-            if not self._relation_created(required_relation):
-                self.unit.status = BlockedStatus(
-                    f"Waiting for `{required_relation}` relation to be created"
-                )
-                return
+
+        if not self.model.relations.get(SDCORE_MANAGEMENT_RELATION_NAME):
+            self.unit.status = BlockedStatus(
+                f"Waiting for `{SDCORE_MANAGEMENT_RELATION_NAME}` relation to be created"
+            )
+            return
         if not self._sdcore_management.management_url:
             self.unit.status = WaitingStatus("Waiting for webui management url to be available")
             event.defer()
             return
-        if not self._get_upf_hostname() or not self._get_upf_port():
+        if not self._fiveg_n4_information_is_correct():
             self.unit.status = WaitingStatus("Waiting for UPF information to be available")
             event.defer()
             return
+        if not self.fiveg_gnb_identity_information_is_correct():
```

Comment:
If there are 12 different relations with 12 different gnb's and one of them does not contain the appropriate information, the nms charm will go to waiting status and never get configured. I don't think that's what we want. Here I don't think we should be validating that the gnb info is correct before configuring pebble. What we may want to do though is only insert correct config data in the config file and if one of the relation is not, log a warning.